chain/network/src/peer_manager/peer_store.rs

```diff
@@ -177,20 +177,17 @@ impl PeerStore {
         }
     }
 
-    fn find_peers<F>(&self, mut filter: F, count: u32) -> Vec<PeerInfo>
+    /// Find a random subset of peers based on filter.
+    fn find_peers<F>(&self, filter: F, count: usize) -> Vec<PeerInfo>
     where
-        F: FnMut(&KnownPeerState) -> bool,
+        F: FnMut(&&KnownPeerState) -> bool,
     {
-        let mut peers = self
-            .peer_states
-            .values()
-            .filter_map(|p| if filter(p) { Some(p.peer_info.clone()) } else { None })
-            .collect::<Vec<_>>();
-        if count == 0 {
-            return peers;
+        let peers = self.peer_states.values().filter(filter).map(|p| &p.peer_info);
+        if count == 0 || count == usize::MAX {
+            return peers.cloned().collect();
         }
-        peers.shuffle(&mut thread_rng());
-        peers.iter().take(count as usize).cloned().collect::<Vec<_>>()
+        let peers: Vec<&PeerInfo> = peers.collect();
+        peers.choose_multiple(&mut thread_rng(), count).cloned().cloned().collect()
```

Comment:
I see that our behavior changed a little bit -- (for case when "count" > peers.len()), right ?